### ISSUE TITLE
Sidebar sizes - icons and font - are way too big 

### ISSUE THREAD

**Issue by FMorschel** [flutter/devtools#8243]
Really not sure what is going on here but the sizes are way too big and I can't scroll down here to see everything (second and third image).

25" ultra-wide monitor (~60cm x ~30cm).
Windows machine, 2560px x 1080px - scale 100%.

![image](https://github.com/user-attachments/assets/3c0c09dd-965c-4cfd-b1fa-46a53f308a1a)
![image](https://github.com/user-attachments/assets/f1c6fabe-6f74-4af5-b281-c77338446552)
![image](https://github.com/user-attachments/assets/5b05e456-0a37-4823-827e-f969cfd58584)

**Comment by FMorschel** [comment]
This also extends to other specific tools like `Logging` (the word behind is Details):

![image](https://github.com/user-attachments/assets/79791288-d28f-4d59-a885-682f6ffe0de9)

**Comment by kenzieschmoll** [comment]
@Piinks @goderbauer did something change with the Flutter framework recently that would affect text styling in this way?

**Comment by goderbauer** [comment]
Nothing comes immediately to mind. Do we know when this approximately started to happen?

**Comment by FMorschel** [comment]
I could not answer that since that I had recently discovered the sidebar (by the Observable Flutter stream) and had not used it yet.

**Comment by hpp0hpp** [comment]
maybe this is caused by the high definition screen, which apply a scale factor  devtools failed to apply the same behavier as the code editor?
![image](https://github.com/user-attachments/assets/05403c76-9c49-434e-a51f-698b3987dc76)

**Comment by FMorschel** [comment]
I don't think so. My scale is 100%. I do have another monitor for 16:9 (the one I mentioned is 21:9) which is 1920x1080 and that monitor does have by default the suggestion for 125% scale. But even when setting it to 100% (both set to this now) and restarting vscode doesn't seem to fix this.

---

That comment did give me another idea though. Can it be because of the two-monitor setup? One (the 16:9) is smaller (in height as well, I mean, not just the ratio fact) than the other, maybe something is not getting calculated correctly here?

**Comment by pndychrg** [comment]
Hi Everyone, 
I'm having the same issues with Scale Resolution set to 100%. Is there any solution for it?

**Comment by FMorschel** [comment]
Could this be happening because I have a two monitor setup with different DPI?

**Comment by elliette** [comment]
I've also been seeing this with the Property Editor embedded in VS Code. I will try to see if I find a difference based on the monitors I'm using.

**Comment by maxmishyn** [comment]
@FMorschel No. It is the same in the single monitor environment too. Both macOS and windows are affected. High DPI and just a FHD (1080) are affected

**Comment by mSaayeh** [comment]
I'm having the same issue with 1080p monitor and 100% scale. Is there any fix for that?

**Comment by kenzieschmoll** [comment]
Another instance of this:
![Screenshot 2025-01-10 at 2 43 22 PM](https://github.com/user-attachments/assets/30fad490-aa65-4694-9ebb-7d3789b5680a)

**Comment by elliette** [comment]
Chatted with @DanTup about this. His theory (which seems very likely) is that we are notifying DevTools of the font size used in VS Code with a query param. From this font size, we then scale up or down various UI elements accordingly. We are probably overly scaling some items when a user has a larger font size set in VS Code.

**Comment by matiazar** [comment]
same here.. any solution??

**Comment by elliette** [comment]
Open question: Should we be scaling any UI elements by the font factor at all?

Currently many (most?) of our UI elements are being scaled by the font factor:
* https://github.com/search?q=repo%3Aflutter%2Fdevtools%20scaleByFontFactor&type=code

Compare this to VS Code, where changing the font size does not scale any of the UI elements:


![Image](https://github.com/user-attachments/assets/3b32210c-40f6-4072-8ca0-ea3dbf0d8b7c)

![Image](https://github.com/user-attachments/assets/091f5e9d-b19d-46e2-bed8-781ac4db4b2c)

**Comment by DanTup** [comment]
>  Should we be scaling any UI elements by the font factor at all?

I'm not sure if you mean in VS Code, or generally. Generally for scaling in VS Code (the UI etc.), you use the zoom settings:

![Image](https://github.com/user-attachments/assets/278719cd-fb3e-49e4-985d-f7ab24fbfbd1)

This also zooms the editor. I think the intention of using the font size in VS Code at all was to try and get a better default for those that might have increased their size for accessibility, however I'm not really sure if that works (because if you need the whole UI larger, you'd probably use the zoom and _not_ the font size).


Perhaps it'd be interesting to remove the scaling that causes some elements to be bigger and get some screenshots (of both the sidebar and some screens like the inspector)?

**Comment by elliette** [comment]
> This also zooms the editor. I think the intention of using the font size in VS Code at all was to try and get a better default for those that might have increased their size for accessibility, however I'm not really sure if that works (because if you need the whole UI larger, you'd probably use the zoom and not the font size).

Yes, after playing around with zooming and changing the font size with VS Code, IntelliJ, and in the browser, here is what I've discovered:

**In browser:**
There is no way for a user to change the font size, but that is okay because native browser zoom also zooms DevTools

**In VS Code:**
1. Changing the zoom: changing the zoom in VS Code increases/decreases the size of everything. It also changes the zoom of the embedded DevTools frames.
2. Changing the font size: Changing the font size in VS Code only changes the font size in the code editor itself, it does not change the font size across the rest of the IDE.

DevTools is currently being passed the font size and scaling to match that.

**In IntelliJ:**
There are three settings that affect the sizing of elements:
1. `Editor > font`: only affects the font size in the code editor itself (the same behavior as VS Code's font size setting)
2. `Appearance > font`: only affects the font size for everywhere other than the code editor itself. Also scales all the buttons, UI elements to match.
3. `Appearance > Zoom`: Behaves the same as `Appearance > font`, increases the font size for everywhere other than the code editor itself. Also scales all the buttons, UI elements to match.

DevTools is currently being passed `Editor > font` and scaling to match that. 

**Proposal:**

Based on this, I think it would make sense to:

1) Do not pass in the font size in VS Code (or ignore it from DevTools). VS Code scales DevTools as expected by zooming. 
2) In IntelliJ, pass either the `Appearance > font` and/or `Appearance > zoom` setting (depending on which are set). DevTools should scale accordingly based on those settings. (I don't know exactly how DevTools is embedded in IntelliJ, but another option might be to set the [devicePixelRatio ](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio)based on the IntelliJ settings to the IFRAME, the `devicePixelRatio` is what Flutter [uses to determine](https://github.com/flutter/engine/pull/33695) the zoom).

**Comment by DanTup** [comment]
> Do not pass in the font size in VS Code (or ignore it from DevTools). VS Code scales DevTools as expected by zooming.

Removing it from VS Code probably makes most sense.. it can ship sooner and avoids confusion from passing something that's not used.

But does that solve the whole problem? Isn't part of the issue that some of the elements are larger than the standard size and looked out of place? (or did that only occur because of the explicit font size?)

**Comment by elliette** [comment]
> Removing it from VS Code probably makes most sense.. it can ship sooner and avoids confusion from passing something that's not used.

SGTM to remove passing form VS Code! @kenzieschmoll is that good with you?

> But does that solve the whole problem? Isn't part of the issue that some of the elements are larger than the standard size and looked out of place? (or did that only occur because of the explicit font size?)

I think that's only because we are applying `scaleByFontFactor` to some UI elements but not others. So some aren't being scaled at all, whereas others are being scaled to the font size passed in by the editor. (So we will still need to fix this for IntelliJ. since zooming in IntelliJ doesn't zoom DevTools).

**Comment by kenzieschmoll** [comment]
> SGTM to remove passing the font size from VS Code! @kenzieschmoll is that good with you?

Yes sounds good. We can leave all the `scaleByFontFactor` code in DevTools for now since we may still need this for IntelliJ / Android Studio.

> I don't know exactly how DevTools is embedded in IntelliJ, but another option might be to set the [devicePixelRatio](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio)based on the IntelliJ settings to the IFRAME, the devicePixelRatio is what Flutter https://github.com/flutter/engine/pull/33695 the zoom.

Is this what is done in VS code? @jwren  do you know if we can do something similar with jx browser?

**Comment by elliette** [comment]
Oh hmm, I skimmed the MDN Web docs, `devicePixelRatio` is read-only so we wouldn't be able to change it. I do think it's possible we might be able to solve this by modifying the CSS of the IFRAME in intellij.

**Comment by DanTup** [comment]
I've [removed the fontSize parameter from VS Code](https://github.com/Dart-Code/Dart-Code/issues/5406). I also compared the default font size in the inspector tree to a number of other different labels in the VS Code UI and confirmed that they are approximately the same size by default (and scale together):

![Image](https://github.com/user-attachments/assets/8b79ba4f-b898-4302-8512-94f9c6bbf307)

**Comment by elliette** [comment]
> I've https://github.com/Dart-Code/Dart-Code/issues/5406. I also compared the default font size in the inspector tree to a number of other different labels in the VS Code UI and confirmed that they are approximately the same size by default (and scale together)

Thanks @DanTup ! I'm preparing a similar change in IntelliJ so hopefully we can remove all the scaling logic in DevTools

**Comment by kenzieschmoll** [comment]
Update from @elliette: This is partially resolved but is blocked on the next version of the IntelliJ plugin being released. The next step once IntelliJ is released is to go through DevTools and clean up all uses of `sizeByFontFactor`.

**Comment by River-Szasz** [comment]
@kenzieschmoll thank you for the updates. Do you know what is the ETA for stable?

**Comment by elliette** [comment]
Hi @River-Szasz - if you are using VS Code, the fix should already be in [version 3.106](https://dartcode.org/releases/v3-106/) or later. For Android Studio/IntelliJ it will be included in the next release.
